lib/thor/actions/create_file.rb

```diff
@@ -43,7 +43,7 @@ def initialize(base, destination, data, config = {})
       # Boolean:: true if it is identical, false otherwise.
       #
       def identical?
-        exists? && File.binread(destination) == render
+        exists? && File.binread(destination) == String.new(render).force_encoding("ASCII-8BIT")
```

Comment:
I'd add a comment here indicating that `binread` defaults to this encoding to explain why this works.